### ISSUE TITLE
Two instances of new-relic browser app targeting diferents accounts are colliding

### ISSUE THREAD

**Issue by Khaos93** [newrelic/newrelic-browser-agent#818]
two instances of new-relic browser app targeting diferents accounts are colliding

## Description

I am part of a team developing a widget that gets integrated into various websites. To monitor the widget's performance, we employ New Relic MicroAgents usint this library available at https://www.npmjs.com/package/@newrelic/browser-agent. However, a challenge arises as this widget is loaded onto pages that already have the New Relic Browser tracking code embedded (using the JS snippet provided by New Relic in the application settings). Consequently, when clients load the widget, it appears to collide with the existing MicroAgent instance, causing the New Relic tracking of their websites stop sending data to this new relic and start sending data to our account it. it seems that the microAgent is overriding the information  of the other agent

Is there a solution to circumvent this issue and enable multiple New Relic agents to target different New Relic accounts simultaneously on the same website?

## Steps to Reproduce

 - use a JS snippet code provided by new relic for browser applications and execute in a script targeting to new-relic account A
 - load a MicroAgent instance  targeting to new-relic account B
 - 

## Expected Behavior

- every script should send the data to it's respective account

## Relevant Logs / Console output

[TIP]:  # ( Include as many relevant details about your environment as possible. )

* ex: BrowserApp
* ex: chrome last version

**Comment by Khaos93** [comment]
npm package is using: 1.238.0
JS snippet browser agent: nr-loader-spa-1221.min.js

**Comment by patrickhousley** [comment]
Thanks for the additional info @Khaos93 . We have made a few changes in the agent since 1221 that could help with this issue. Is there any way you could try upgrading the agent that is injected on the page to the latest version?

**Comment by Khaos93** [comment]
we already update the agent to 1.247 the version that new-relic recommend us in the browser-application but the issue is still happening

**Comment by patrickhousley** [comment]
Thanks @Khaos93 . Just to confirm, you tried using 1.247 for both the page agent and the micro agent? If so, I will try to look into this as soon as possible.

**Comment by Khaos93** [comment]
yes, the test was using the 1.247 version for both the page agent and the micro agent, thanks

**Comment by patrickhousley** [comment]
Hi @Khaos93 . I was able to setup a test case locally using https://github.com/mits-gossau/event-driven-web-components-realworld-example-app as a test base project. The page loads a full agent and the article page component loads a micro agent. In our testing, we do not see the page level agent changed to start reporting to the micro agent app ID. Is it possible for you to provide a small reproduction of this issue?

**Comment by shaun0024** [comment]
> Hi @Khaos93 . I was able to setup a test case locally using https://github.com/mits-gossau/event-driven-web-components-realworld-example-app as a test base project. The page loads a full agent and the article page component loads a micro agent. In our testing, we do not see the page level agent changed to start reporting to the micro agent app ID. Is it possible for you to provide a small reproduction of this issue?


@patrickhousley wondering if you might have the test code you used against this https://github.com/mits-gossau/event-driven-web-components-realworld-example-app repo that you can share?

**Comment by patrickhousley** [comment]
Hey @shaun0024 let me reopen this and recreate the code for you.

**Comment by shaun0024** [comment]
> Hey @shaun0024 let me reopen this and recreate the code for you.

Thank you @patrickhousley , much appreciated

**Comment by patrickhousley** [comment]
@shaun0024 Please find attached the example app. Just use `python3 -m http.server` in the root directory to start the server. The `src/index.html` file contains a a full agent using copy/paste and loading assets from our CDN while the component `src/es/components/molecules/TagList.js` has a micro agent.

When you load the page, you will get data for the full agent in the page and a PageView and JSError from the micro agent. If you switch tabs and come back, you should see some INS calls (insights). You will need to update both agents with valid license keys and settings from New Relic.

[event-driven-web-components-realworld-newrelic-test.zip](https://github.com/newrelic/newrelic-browser-agent/files/14216155/event-driven-web-components-realworld-newrelic-test.zip)

**Comment by shaun0024** [comment]
> @shaun0024 Please find attached the example app. Just use `python3 -m http.server` in the root directory to start the server. The `src/index.html` file contains a a full agent using copy/paste and loading assets from our CDN while the component `src/es/components/molecules/TagList.js` has a micro agent.
> 
> When you load the page, you will get data for the full agent in the page and a PageView and JSError from the micro agent. If you switch tabs and come back, you should see some INS calls (insights). You will need to update both agents with valid license keys and settings from New Relic.
> 
> [event-driven-web-components-realworld-newrelic-test.zip](https://github.com/newrelic/newrelic-browser-agent/files/14216155/event-driven-web-components-realworld-newrelic-test.zip)

Thank you very much @patrickhousley!

**Comment by patrickhousley** [comment]
@shaun0024 I am going to close this ticket due to the lack of engagement. If there is a bug or issue you would like us to look at, please feel free to open a new ticket.
